src/test/java/org/janelia/saalfeldlab/n5/hdf5/N5HDF5Test.java

```diff
@@ -335,4 +361,211 @@ public void testAttributesAsJson() throws IOException {
 	public void testType() {
 		System.out.println(new TypeToken<DataType>() {}.getType().getTypeName());
 	}
+	
+	@Override
+	@Test
+	/*
+	 * Differs from AbstractN5Test since an int will be read back as int, not a long
+	 */
+	public void testListAttributes() {
+
+		final String groupName2 = groupName + "-2";
+		final String datasetName2 = datasetName + "-2";
+		try {
+			n5.createDataset(datasetName2, dimensions, blockSize, DataType.UINT64, new RawCompression());
+			n5.setAttribute(datasetName2, "attr1", new double[] {1.1, 2.1, 3.1});
+			n5.setAttribute(datasetName2, "attr2", new String[] {"a", "b", "c"});
+			n5.setAttribute(datasetName2, "attr3", 1.1);
+			n5.setAttribute(datasetName2, "attr4", "a");
+			n5.setAttribute(datasetName2, "attr5", new long[] {1, 2, 3});
+			n5.setAttribute(datasetName2, "attr6", 1);
+			n5.setAttribute(datasetName2, "attr7", new double[] {1, 2, 3.1});
+			n5.setAttribute(datasetName2, "attr8", new Object[] {"1", 2, 3.1});
+
+			Map<String, Class<?>> attributesMap = n5.listAttributes(datasetName2);
+			Assert.assertTrue(attributesMap.get("attr1") == double[].class);
+			Assert.assertTrue(attributesMap.get("attr2") == String[].class);
+			Assert.assertTrue(attributesMap.get("attr3") == double.class);
+			Assert.assertTrue(attributesMap.get("attr4") == String.class);
+			Assert.assertTrue(attributesMap.get("attr5") == long[].class);
+			//HDF5 will parse an int as an int rather than a long
+			Assert.assertTrue(attributesMap.get("attr6") == int.class);
```

Comment:
This differs from `AbstractN5Test`, which expects a `long`.

Reply:
Great!